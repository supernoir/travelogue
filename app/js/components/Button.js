import React from 'react';

export default class Button extends React.Component {
	constructor() {
		super();
		this.doSomething = this.doSomething.bind(this);
	}

	doSomething() {
		console.log('something');
	}

	render() {
		return (
			<a href="#" className={'btn'} id={'button'} onClick={this.doSomething.bind(this)}>
				{this.props.label}
			</a>
		);
	}
}
