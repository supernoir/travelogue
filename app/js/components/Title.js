import React from 'react';
import PropTypes from 'prop-types';

export default class Title extends React.Component {
	render() {
		return <h1>{this.props.headline}</h1>;
	}
}

Title.propTypes = {
	headline: PropTypes.string
};
