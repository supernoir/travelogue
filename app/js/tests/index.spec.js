import React from 'react';

import { shallow } from 'enzyme';
import sinon from 'sinon';

import Title from '../components/Title';
import Button from '../components/Button';

test('Renders a Title', () => {
	const wrapper = shallow(<Title headline={'I am a Title'} />);
	expect(wrapper).toMatchSnapshot();
});

test('Renders a Title with prop title', () => {
	const wrapper = shallow(<Title headline={'Example'} />);
	expect(wrapper.prop('headline')).toEqual('Example');
});

test('Simulate Button clicked', () => {
	const wrapper = shallow(<Button label={'Click me'} />);
	expect(wrapper).toMatchSnapshot();
	wrapper.find('.btn').simulate('click');
	expect(wrapper).toMatchSnapshot();
});

describe('<MyComponent />', () => {
	it('simulates click events', () => {
		const onButtonClick = sinon.spy();
		const wrapper = shallow(<Button />);
		wrapper.find('.btn').simulate('click');
		expect(onButtonClick.calledOnce).to.equal(true);
	});
});
