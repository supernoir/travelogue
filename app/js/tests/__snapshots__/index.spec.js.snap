// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a Title 1`] = `
<h1>
  I am a Title
</h1>
`;

exports[`Simulate Button clicked 1`] = `
<a
  className="btn"
  href="#"
  id="button"
  onClick={[Function]}
>
  Click me
</a>
`;

exports[`Simulate Button clicked 2`] = `
<a
  className="btn"
  href="#"
  id="button"
  onClick={[Function]}
>
  Click me
</a>
`;
